tests/Backend/ExternalBuckets/BigqueryRegisterBucketTest.php

```diff
@@ -237,6 +237,124 @@ public function testRefreshBucketWhenSchemaDoesNotExist(): void
         }
     }
 
+    /**
+     * @group noSOX
+     * @dataProvider provideComponentsClientTypeBasedOnSuite
+     */
+    public function testAlteredColumnThrowsUserExAndAfterRefreshWillWork(): void
+    {
+        $description = $this->generateDescriptionForTestObject();
+        $testBucketName = $this->getTestBucketName($description);
+        $bucketId = self::STAGE_IN . '.' . $testBucketName;
+
+        $this->dropBucketIfExists($this->_client, $bucketId, true);
+
+        // prepare external bucket
+        $path = $this->prepareExternalBucketForRegistration($description);
+
+        $externalCredentials['connection']['backend'] = 'bigquery';
+        $externalCredentials['connection']['credentials'] = $this->getCredentialsArray();
+        $schemaName = $externalCredentials['connection']['schema'] = sha1($description) . '_external_bucket';
+
+        $db = WorkspaceBackendFactory::createWorkspaceBackend($externalCredentials);
+
+        $db->createTable('MY_LITTLE_TABLE_FOR_VIEW', ['AMOUNT' => 'INT64', 'DESCRIPTION' => 'STRING']);
+
+        $db->executeQuery(sprintf(
+        /** @lang BigQuery */
+            'CREATE VIEW `%s`.`MY_LITTLE_VIEW` AS SELECT * FROM `%s`.`MY_LITTLE_TABLE_FOR_VIEW`',
+            $schemaName,
+            $schemaName,
+        ));
+
+        $idOfBucket = $this->_client->registerBucket(
+            $testBucketName,
+            $path,
+            'in',
+            'Iam in workspace',
+            'bigquery',
+            'Iam-your-external-bucket-to-test-alter-table',
+        );
+
+        // check table existence
+        $tables = $this->_client->listTables($idOfBucket);
+        $this->assertCount(2, $tables);
+        $table = $tables[0];
+        $view = $tables[1];
+
+        $db->executeQuery(sprintf(
+        /** @lang BigQuery */
+            'ALTER TABLE `%s`.`MY_LITTLE_TABLE_FOR_VIEW` ADD COLUMN AGE INTEGER;',
+            $schemaName,
+        ));
+
+        $db->executeQuery(sprintf(
+        /** @lang BigQuery */
+            'CREATE OR REPLACE VIEW `%s`.`MY_LITTLE_VIEW` AS SELECT * FROM `%s`.`MY_LITTLE_TABLE_FOR_VIEW`',
+            $schemaName,
+            $schemaName,
+        ));
+
+        /** @var array $tableDataPreview */
+        $tableDataPreview = $this->_client->getTableDataPreview($table['id'], ['format' => 'json']);
+        /** @var array $viewDataPreview1 */
+        $viewDataPreview1 = $this->_client->getTableDataPreview($view['id'], ['format' => 'json']);
+        $this->assertSame(['AMOUNT', 'DESCRIPTION'], $tableDataPreview['columns']);
+        $this->assertSame(['AMOUNT', 'DESCRIPTION'], $viewDataPreview1['columns']);
```

Comment:
sorry myslel som ze som to tam poslal ale este som robil lepsiu validaciu tak ako to je u snowflaku